### PR TITLE
[RPD-287] ZenML version inference for zenserver

### PR DESCRIPTION
**Pull request by KirsoppJ**
This PR has matcha look for a local install of zenml in the environment from which it was launched, then sets the version to be installed on the remote resources accordingly. If no installation is found, then the most recent release of zen server is installed instead.

Most of the changes are to terraform files, where a new variable has been introduced. On the python side, the version is passed to `build_template_configuration` after it is inferred by new `infer_zenml_version` function. This function also prints to the terminal information regarding the selected zenml version.

## Checklist

Please ensure you have done the following:

* [x] I have read the [CONTRIBUTING](https://github.com/fuzzylabs/matcha/blob/main/CONTRIBUTING.md) guide.
* [ ] I have updated the documentation if required.
* [ ] I have added tests which cover my changes.

## Type of change

Tick all those that apply:

* [ ] Bug Fix (non-breaking change, fixing an issue)
* [x] New feature (non-breaking change to add functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [ ] Other (add details above)

### CODE REVIEW

File: src/matcha_ml/infrastructure/resources/zen_server/main.tf [hunk 1]
```diff
@@ -86,6 +86,10 @@ resource "helm_release" "zen_server" {
     name  = "zenml.database.sslVerifyServerCert"
     value = var.deploy_db ? false : var.database_ssl_verify_server_cert
   }
+  set {
+    name = "zenml.version"
```
Comment (dudeperf3ct) [inline src/matcha_ml/infrastructure/resources/zen_server/main.tf:90]: This should be `zenml.image.tag`

**Comment by JonoCX** [comment]
It'd be good if this branch could be merged into an improvement branch: `fixes/` or similar

File: src/matcha_ml/core/core.py [hunk 2]
```diff
@@ -203,7 +202,7 @@ def provision(
         MatchaError: If prefix is not valid.
         MatchaError: If region is not valid.
     """
-    zenml_version_is_supported()
+    # zenml_version_is_supported()
```
Comment (JonoCX) [inline src/matcha_ml/core/core.py:205]: No longer needed?

**Review by JonoCX — CHANGES_REQUESTED** [review]
LGTM - one minor change to remove old code

**Pull request merged**